test/images/image-util.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash -x
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TASK=$1
+TEST=$2
+ARCH=${3:-all}
+
+# Mapping of go ARCH to actual architectures shipped part of multiarch/qemu-user-static project
+declare -A QEMUARCHS=( ["amd64"]="x86_64" ["arm"]="arm" ["arm64"]="aarch64" ["ppc64le"]="ppc64le" ["s390x"]="s390x" )
+
+# Returns list of all supported architectures from BASEIMAGE file
+listArchs() {
+  (
+    cut -d "=" -f 1 ${TEST}/BASEIMAGE
+  )
+}
+
+# Returns baseimage need to used in Dockerfile for any given architecture
+getBaseImage() {
+  arch=$1
+  echo $(grep "${arch}=" BASEIMAGE | cut -d= -f2)
+}
+
+# This function will build test image for all the architectures
+# mentioned in BASEIMAGE file. In the absence of BASEIMAGE file,
+# it will build for all the supported arch list - amd64, arm,
+# arm64, ppc64le, s390x
+build() {
+  if [[ -f ${TEST}/BASEIMAGE ]]; then
+    archs=$(listArchs)
+  else
+    archs=${!QEMUARCHS[@]}
+  fi
+
+  for arch in ${archs}; do
+    echo "Build image for ${TEST} ARCH: ${arch}..."
+    temp_dir=$(mktemp -d)
+    cp -r ${TEST}/* ${temp_dir}
+    if [[ -f ${TEST}/bootstrap.sh ]]; then
+      pushd ${TEST}
+      ./bootstrap.sh ${arch} ${temp_dir}
+      popd
+    fi
+    pushd ${temp_dir}
+    TAG=$(<VERSION)
+    IMAGENAME=$(<NAME)
+
+    if [[ -f BASEIMAGE ]]; then
+      BASEIMAGE=$(getBaseImage ${arch})
+      sed -i.bak "s|BASEIMAGE|${BASEIMAGE}|g" Dockerfile
+    fi
+
+    if [[ $(grep "CROSS_BUILD_" Dockerfile) ]]; then
+      if [[ "${arch}" == "amd64" ]]; then
+        sed -i "/CROSS_BUILD_/d" Dockerfile
+      else
+        sed -i "s|ARCH|${QEMUARCHS[$arch]}|g" Dockerfile
+        docker run --rm --privileged multiarch/qemu-user-static:register --reset
+        curl -sSL https://github.com/multiarch/qemu-user-static/releases/download/${QEMUVERSION}/x86_64_qemu-${QEMUARCHS[$arch]}-static.tar.gz | tar -xz -C ${temp_dir}
+        sed -i "s/CROSS_BUILD_//g" Dockerfile
+      fi
+    fi
+
+    docker build --pull -t ${REGISTRY}/${IMAGENAME}-${arch}:${TAG} .
+
+    # Image without any arch postfix will point to amd64 by default.
```

Comment:
* `gcr.io/kubernetes-e2e-test-images` exists now
* I'm in favor of pushing only the `-$arch` versions and removing the legacy amd64 workaround.
* directory name = image name seems like a good idea. it doesn't necessarily need to happen in this PR.
* agree that probably standardizing tags is a good idea. I don't have a strong opinion on exactly what format to use.